File: EmployeeService.Api/Contracts/Models/Employee.cs
namespace EmployeeService.Api.Contracts.Models;

public record Employee
{
    public int Id { get; set; }
    
    public string Name { get; set; }

    public string Surname { get; set; }
    
    public string Phone { get; set; }
    
    public int CompanyId { get; set; }
    
    public Passport Passport { get; set; }
    
    public Department Department { get; set; }
}

File: EmployeeService.WebInterface/src/app/contracts/employee.ts
import {Department} from './department';
import {Passport} from './passport';

export class Employee {
	public id: number;

	public name: string;

	public surname: string;

	public phone: string;

	public companyId: number;

	public department: Department;

	public passport: Passport;

	public constructor(
		id: number,
		name: string,
		surname: string,
		phone: string,
		companyId: number,
		department: Department,
		passport: Passport) {
		this.id = id;
		this.name = name;
		this.surname = surname;
		this.phone = phone;
		this.companyId = companyId;
		this.department = department;
		this.passport = passport;
	}
}
